compiler/z/codegen/OMRTreeEvaluator.cpp

```diff
@@ -14116,6 +14116,195 @@ bool isSettingOp(uint8_t byteValue, TR::InstOpCode::Mnemonic SI_opcode)
       return false;
    }
 
+TR::Register *
+OMR::Z::TreeEvaluator::bitpermuteEvaluator(TR::Node *node, TR::CodeGenerator *cg)
+   {
+   TR::Node *value = node->getChild(0);
+   TR::Node *addr = node->getChild(1);
+   TR::Node *length = node->getChild(2);
+
+   auto valueReg = cg->evaluate(value);
+   auto addrReg = cg->evaluate(addr);
+
+   TR::Register *tmpReg = cg->allocateRegister(TR_GPR);
+
+   static const int8_t BIT_PERMUTE_CONSTANT_UNROLL_THRESHOLD = 4;
+
+   TR::Register *resultReg = cg->allocateRegister(TR_GPR);
+
+
+   if (length->getOpCode().isLoadConst() && length->getIntegerNodeValue<int32_t>() <= BIT_PERMUTE_CONSTANT_UNROLL_THRESHOLD)
+      {
+      // Manage the constant length case
+      int32_t arrayLen = length->getIntegerNodeValue<int>();
```

Comment:
Can we pull this outside of the above if and use `arrayLen` instead of `length->getIntegerNodeValue<int32_t>()`? Also this line seems to be extracting an `int` but the type of the variable is `int32_t`. Can we fix this?

Reply:
Addressed here: 95356e13b0e6942385bcc67e44f1876009725b6e